app/models/potential_matcher/prompt_tag_type_info.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+# A class used for gathering information about the rough frequency of different
+# required tag types in all of the prompts in a collection. Used to figure out
+# what would be a good "index type" -- that is, a type that we can use to try
+# to look up matches with.
+class PromptTagTypeInfo
```

Comment:
QA note: run this against all real-world challenges in the db on staging in console to see what index types result just as sanity check, looks good to me